WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipping Labels/Create Shipping Label Form/ShippingLabelFormViewModel.swift

```diff
@@ -12,8 +12,15 @@ final class ShippingLabelFormViewModel {
     let destinationAddress: ShippingLabelAddress?
 
     init(originAddress: Address?, destinationAddress: Address?) {
+        let accountSettings = ShippingLabelFormViewModel.getStoredAccountSettings()
+        let company = ServiceLocator.stores.sessionManager.defaultSite?.name
+        let defaultAccount = ServiceLocator.stores.sessionManager.defaultAccount
+
         self.originAddress = ShippingLabelFormViewModel.fromAddressToShippingLabelAddress(address: originAddress) ??
-            ShippingLabelFormViewModel.getDefaultOriginAddress()
+            ShippingLabelFormViewModel.getDefaultOriginAddress(accountSettings: accountSettings,
+                                                               company: company,
+                                                               siteAddress: SiteAddress(),
+                                                               account: defaultAccount)
```

Comment:
nit: maybe we could have some unit tests on the initial origin and destination address now that there is quite some logic involved?

Reply:
I started implementing them in a later PR 😅